lib/JSONSchemaGenerator.js

```diff
@@ -63,39 +63,108 @@ const {
 
 const EXAMPLE_DATE = new Date('2020-09-16T12:00:00Z').toISOString()
 const mandatoryFields = new Set([MONGOID, UPDATERID, UPDATEDAT, CREATORID, CREATEDAT, __STATE__])
-const mandatoryFieldsWithoutId = (isNewSchema) => ({
-  [CREATORID]: {
-    type: 'string',
-    description: 'User id that has created this object',
-  },
-  [CREATEDAT]: {
-    type: 'string',
-    ...isNewSchema ? { format: 'date-time' } : { pattern: '^\\d{4}-\\d{2}-\\d{2}(T\\d{2}:\\d{2}:\\d{2}(\\.\\d{1,3})?(Z|[+-]\\d{2}:\\d{2}))?$' },
-    description: 'Date of the request that has performed the object creation',
-    examples: [
-      '2020-09-16T12:00:00.000Z',
-    ],
-  },
-  [UPDATERID]: {
-    type: 'string',
-    description: 'User id that has requested the last change successfully',
-  },
-  [UPDATEDAT]: {
-    type: 'string',
-    ...isNewSchema ? { format: 'date-time' } : { pattern: '^\\d{4}-\\d{2}-\\d{2}(T\\d{2}:\\d{2}:\\d{2}(\\.\\d{1,3})?(Z|[+-]\\d{2}:\\d{2}))?$' },
-    description: 'Date of the request that has performed the last change',
-    examples: [
-      '2020-09-16T12:00:00.000Z',
-    ],
-  },
-})
+const mandatoryFieldsWithoutId = (isNewSchema, validate) => {
```

Comment:
Consider this as a suggestion: sooner or later we should split the content of this file (it's almost 2k lines), and move most of these getter functions in a different file to improve the readability of this file (we don't need to keep an eye on the schema of UPDATEDAT every time we need to review of a specific Schema is generated).

My take would be to create a file (`schemaGetters.js`) and include there all the methods/consts that returns piece of schemas (everything before the definition of the JsonSchemaGenerator class) and a second file (`JSONSchemaGenerator.utils.js`) to include all the remaining functions outside the class (this last one can be postponed to a refactor task because we can consider to add some unit tests).

Reply:
Done